hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -3926,20 +3927,26 @@ public Boolean isTransient(Object entity, SharedSessionContractImplementor sessi
 			// let this take precedence if defined, since it works for
 			// assigned identifiers
 			final Object version = getVersion( entity );
-			final Boolean result = versionMapping.getUnsavedStrategy().isUnsaved( version );
-			if ( result != null ) {
-				if ( result && version == null && session.getPersistenceContext().hasLoadContext() ) {
-					// check if we're currently loading this entity instance, the version
-					// will be null but the entity cannot be considered transient
-					final EntityKey entityKey = new EntityKey( id, this );
-					final LoadingEntityEntry loadingEntityEntry = session.getPersistenceContext()
-							.getLoadContexts()
-							.findLoadingEntityEntry( entityKey );
-					if ( loadingEntityEntry != null && loadingEntityEntry.getEntityInstance() == entity ) {
-						return false;
+			final Boolean isUnsaved = versionMapping.getUnsavedStrategy().isUnsaved( version );
+			if ( isUnsaved != null ) {
+				if ( isUnsaved  ) {
+					if ( version == null && session.getPersistenceContext().hasLoadContext() ) {
+						// check if we're currently loading this entity instance, the version
+						// will be null but the entity cannot be considered transient
+						final LoadingEntityEntry loadingEntityEntry = session.getPersistenceContext()
+								.getLoadContexts()
+								.findLoadingEntityEntry( new EntityKey( id, this ) );
+						if ( loadingEntityEntry != null && loadingEntityEntry.getEntityInstance() == entity ) {
+							return false;
+						}
+					}
+					final Boolean unsaved = identifierMapping.getUnsavedStrategy().isUnsaved( id );
+					if ( unsaved != null && !unsaved ) {
+						throw new VersionValueException(
+								"A Persisted Versioned Entity cannot have an unsaved version value" );
```

Comment:
How about:

    "Transient entity with generated id '" + id + "' has an uninitialized value '" + version + "' for its version attribute"

Reply:
Perhaps even just:

> "Detached entity with generated id '" + id + "' has uninitialized version value '" + version + "'"